packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/MrwSerializationTypeDefinition.cs

```diff
@@ -1384,6 +1632,42 @@ private MethodBodyStatement CreateWriteAdditionalRawDataStatement()
             };
         }
 
+        private MethodBodyStatement CreateWriteAdditionalPropertiesStatement()
+        {
+            if (_inputModel.AdditionalProperties == null || AdditionalProperties.Length == 0)
+            {
+                return MethodBodyStatement.Empty;
+            }
+
+            var statements = new MethodBodyStatement[AdditionalProperties.Length];
+            for (int i = 0; i < AdditionalProperties.Length; i++)
+            {
+                var additionalPropertiesProperty = AdditionalProperties[i];
+                var tKey = additionalPropertiesProperty.Type.Arguments[0];
+                var tValue = additionalPropertiesProperty.Type.Arguments[1];
+                // generate serialization statements for each key-value pair in the additional properties dictionary
+                var forEachStatement = new ForeachStatement("item", additionalPropertiesProperty.AsDictionary(tKey, tValue), out KeyValuePairExpression item)
+                {
+                    _utf8JsonWriterSnippet.WritePropertyName(item.Key),
+                    CreateSerializationStatement(additionalPropertiesProperty.Type.Arguments[1], item.Value, SerializationFormat.Default),
+                };
+                statements[i] = forEachStatement;
+            }
+
+            return statements;
+        }
+
+        private PropertyProvider? GetAdditionalBinaryDataPropertiesProp()
+        {
+            PropertyProvider? property = _model.Properties.FirstOrDefault(
+                p => p.Name == AdditionalPropertiesHelper.AdditionalBinaryDataPropsFieldName
```

Comment:
Why do we need the OR here? When would we have one condition but not the other?

Reply:
Good callout, I must've overlooked this. We can just check if the backing field is set to the binary data field.